### PR TITLE
my back for my to-do list

### PR DESCRIPTION
**Pull request by tolyanchenko**

### CODE REVIEW

File: src/modules/controllers/task-controllers.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+const Task = require("../../db/models/task/index");
+
+module.exports.getTasks = (req, res) => {
+  Task.find().then((result) => {
+    res.send({ data: result });
+  });
+};
+
+module.exports.createTask = (req, res) => {
+  const task = new Task(req.body);
```
Comment (usatovaDasha) [inline src/modules/controllers/task-controllers.js:10]: if i send empty body?

File: src/modules/controllers/task-controllers.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  task
+    .save()
+    .then((result) => {
+      res.send(result);
+    })
+    .catch((err) => console.log(err));
+};
+
+module.exports.updateTask = (req, res) => {
+  Task.updateOne({ _id: req.body.id }, req.body).then((result) => {
```
Comment (usatovaDasha) [inline src/modules/controllers/task-controllers.js:20]: if i don't send id and empty body?

File: src/modules/controllers/task-controllers.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    Task.find({ _id: req.body.id }).then((result) => {
+      res.send("The task was changed");
+    });
+  });
+};
+
+module.exports.deleteTask = (req, res) => {
+  Task.deleteOne({ _id: req.query.id }).then((result) => {
```
Comment (usatovaDasha) [inline src/modules/controllers/task-controllers.js:28]: if i don't send id ?

File: app.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+const express = require("express");
+// const bodyParser = require("body-parser");
```
Comment (usatovaDasha) [inline app.js:2]: remove comments

File: src/modules/controllers/task-controllers.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
+const Task = require("../../db/models/task/index");
+
+module.exports.getTasks = (req, res) => {
+  Task.find().then((result) => {
+    res.send({ data: result });
+  });
+};
```
Comment (usatovaDasha) [inline src/modules/controllers/task-controllers.js:7]: after } ; not put

File: src/modules/controllers/task-controllers.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+module.exports.createTask = (req, res) => {
+  const task = new Task(req.body);
+  task
+    .save()
+    .then((result) => {
+      res.send(result);
+    })
+    .catch((err) => console.log(err));
+};
```
Comment (usatovaDasha) [inline src/modules/controllers/task-controllers.js:17]: after } ; not put

File: src/modules/controllers/task-controllers.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+module.exports.updateTask = (req, res) => {
+  Task.updateOne({ _id: req.body.id }, req.body).then((result) => {
+    Task.find({ _id: req.body.id }).then((result) => {
+      res.send("The task was changed");
+    });
+  });
+};
```
Comment (usatovaDasha) [inline src/modules/controllers/task-controllers.js:25]: after } ; not put

File: src/modules/controllers/task-controllers.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+module.exports.deleteTask = (req, res) => {
+  Task.deleteOne({ _id: req.query.id }).then((result) => {
+    res.send("The task was deleted");
+  });
+};
```
Comment (usatovaDasha) [inline src/modules/controllers/task-controllers.js:31]: after } ; not put

File: src/modules/controllers/task-controllers.js (hunk 6 above)
Comment (tolyanchenko) [reply src/modules/controllers/task-controllers.js:17]: But VS code does it itself, really! It's not like I really wanted to be doing it myself...

File: src/modules/controllers/task-controllers.js [hunk 9]
```diff
@@ -8,24 +8,36 @@ module.exports.getTasks = (req, res) => {
 
 module.exports.createTask = (req, res) => {
   const task = new Task(req.body);
-  task
-    .save()
-    .then((result) => {
+  if (req.body.hasOwnProperty("text") && req.body.hasOwnProperty("isCheck")) {
```
Comment (usatovaDasha) [inline src/modules/controllers/task-controllers.js:11]: if i send empty data?

File: src/modules/controllers/task-controllers.js [hunk 10]
```diff
@@ -8,24 +8,36 @@ module.exports.getTasks = (req, res) => {
...
 module.exports.updateTask = (req, res) => {
-  Task.updateOne({ _id: req.body.id }, req.body).then((result) => {
-    Task.find({ _id: req.body.id }).then((result) => {
-      res.send("The task was changed");
+  if (
+    (req.body.hasOwnProperty("text") && req.body.hasOwnProperty("id")) ||
+    (req.body.hasOwnProperty("id") && req.body.hasOwnProperty("isCheck"))
+  ) {
+    Task.updateOne({ _id: req.body.id }, req.body).then((result) => {
+      Task.find({ _id: req.body.id }).then((result) => {
```
Comment (usatovaDasha) [inline src/modules/controllers/task-controllers.js:26]: why need find, if you send text?

File: src/modules/controllers/task-controllers.js [hunk 11]
```diff
@@ -8,24 +8,36 @@ module.exports.getTasks = (req, res) => {
...
+    res.status(422).send("Error! Params not correct");
+  }
 };
 
 module.exports.updateTask = (req, res) => {
-  Task.updateOne({ _id: req.body.id }, req.body).then((result) => {
-    Task.find({ _id: req.body.id }).then((result) => {
-      res.send("The task was changed");
+  if (
+    (req.body.hasOwnProperty("text") && req.body.hasOwnProperty("id")) ||
```
Comment (usatovaDasha) [inline src/modules/controllers/task-controllers.js:22]: higher

File: src/modules/controllers/task-controllers.js (hunk 9 above)
Comment (tolyanchenko) [reply src/modules/controllers/task-controllers.js:11]: Снимок экрана от 2021-08-18 14-39-32

File: src/modules/controllers/task-controllers.js (hunk 11 above)
Comment (tolyanchenko) [reply src/modules/controllers/task-controllers.js:22]: Снимок экрана от 2021-08-18 14-53-42

File: src/modules/controllers/task-controllers.js (hunk 9 above)
Comment (tolyanchenko) [reply src/modules/controllers/task-controllers.js:11]: Снимок экрана от 2021-08-18 14-39-32

File: src/modules/controllers/task-controllers.js (hunk 11 above)
Comment (tolyanchenko) [reply src/modules/controllers/task-controllers.js:22]: Снимок экрана от 2021-08-18 14-57-56

Comment (usatovaDasha) [reply src/modules/controllers/task-controllers.js:22]: ok
{ 
  "_id": "",
  "text": "",
  "isCheck": ""
}

File: src/modules/controllers/task-controllers.js [hunk 12]
```diff
@@ -8,31 +8,44 @@ module.exports.getTasks = (req, res) => {
 
 module.exports.createTask = (req, res) => {
   const task = new Task(req.body);
-  if (req.body.hasOwnProperty("text") && req.body.hasOwnProperty("isCheck")) {
-    task.save().then((result) => {
-      res.send(result);
-    });
+  const { text, isCheck } = task;
+  if (text != "" && (isCheck === true || isCheck === false)) {
```
Comment (usatovaDasha) [inline src/modules/controllers/task-controllers.js:12]: if (req.body.hasOwnProperty("text") && req.body.hasOwnProperty("isCheck") && text && (isCheck === true || isCheck === false)) {

**Pull request merged**